packages/core/strapi/lib/services/entity-validator/validators.js

```diff
@@ -20,69 +20,109 @@ const composeValidators = (...fns) => (attr, { isDraft }) => {
   _.isInteger(minLength) && !isDraft ? validator.min(minLength) : validator;
 
 /**
  * Adds maxLength validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxLengthValidator = ({ maxLength }, validator) =>
+const addMaxLengthValidator = ({ maxLength }, __, validator) =>
   _.isInteger(maxLength) ? validator.max(maxLength) : validator;
 
 /**
  * Adds min integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinIntegerValidator = ({ min }, validator) =>
+const addMinIntegerValidator = ({ min }, __, validator) =>
   _.isNumber(min) ? validator.min(_.toInteger(min)) : validator;
 
 /**
  * Adds max integer validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxIntegerValidator = ({ max }, validator) =>
+const addMaxIntegerValidator = ({ max }, __, validator) =>
   _.isNumber(max) ? validator.max(_.toInteger(max)) : validator;
 
 /**
  * Adds min float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMinFloatValidator = ({ min }, validator) =>
+const addMinFloatValidator = ({ min }, __, validator) =>
   _.isNumber(min) ? validator.min(min) : validator;
 
 /**
  * Adds max float/decimal validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addMaxFloatValidator = ({ max }, validator) =>
+const addMaxFloatValidator = ({ max }, __, validator) =>
   _.isNumber(max) ? validator.max(max) : validator;
 
 /**
  * Adds regex validator
  * @param {Object} attribute model attribute
  * @param {Object} validator yup validator
  */
-const addStringRegexValidator = ({ regex }, validator) =>
+const addStringRegexValidator = ({ regex }, __, validator) =>
   _.isUndefined(regex) ? validator : validator.matches(new RegExp(regex));
 
+const addUniqueValidator = (attr, { model, attributeName, entity, data }, validator) => {
+  /**
+   * If the attribute value is `null` we want to skip the unique validation.
+   * Otherwise it'll only accept a single `null` entry in the database.
+   */
+  if (data === null) {
+    return validator;
+  }
+
+  /**
+   * If the attribute is unchanged we skip the unique verification. This will
+   * prevent the validator to be triggered in case the user activated the
+   * unique constraint after already creating multiple entries with
+   * the same attribute value for that field.
+   */
+  if (entity && data === entity[attributeName]) {
+    return validator;
+  }
+
+  if (attr.unique || attr.type === 'uid') {
+    return validator.test('unique', 'This attribute must be unique', async value => {
+      let whereParams = entity
+        ? { $and: [{ [attributeName]: value }, { $not: { id: entity.id } }] }
+        : { [attributeName]: value };
+
+      const record = await strapi.db.query(model.uid).findOne({
```

Comment:
For later but might be worth it to add an `exists` query as it is a recurring need